File: 2024/Day01/Python/Historian_Hysteria_P1.py
file_name = "input_Historian_Hysteria_P1.txt"

try:
    with open(file_name, "r") as file:
        first_column = []
        second_column = []
        while True:
            line = file.readline()
            if not line:
                break
            x, y = map(int, line.split())
            first_column.append(x)
            second_column.append(y)
        
        first_column.sort()
        second_column.sort()

        Sum = 0
        for i in range(len(first_column)):
            Sum += abs(first_column[i] - second_column[i])
        
        print(Sum)

except:
    print("Error")

File: 2024/Day01/Go/Historian_Hysteria_P1.go
package main

import (
	"fmt"
	"strings"
	"os"
	"bufio"
	"strconv"
	"slices"
	"math"
)


func main() {
	file, err := os.Open("input")
	if err != nil {
		fmt.Println(err)
		return
	}
	scanner := bufio.NewScanner(file)
	
	right := []int{}
  left := []int{}

	for scanner.Scan() {
		line := strings.Split(scanner.Text(), "   ");
		number, _ := strconv.Atoi(line[0])
		left = append(left, number)
		number, _ = strconv.Atoi(line[1])
		right = append(right, number) 
	}

	slices.Sort(left)
	slices.Sort(right)

	distances := make(map[int]int)

	for index, _ := range left {
		distance := int(math.Abs(float64(left[index] - right[index])))
		distances[distance]++;
	}

	var result int

	for key, value := range distances {
		result += (key * value)
	}

	fmt.Println(result)
}
